rclcpp/test/rclcpp/test_any_service_callback.cpp

```diff
@@ -73,6 +74,36 @@ TEST_F(TestAnyServiceCallback, set_and_dispatch_header) {
 
   any_service_callback_.set(callback_with_header);
   EXPECT_NO_THROW(
-    any_service_callback_.dispatch(request_header_, request_, response_));
+    EXPECT_NE(nullptr, any_service_callback_.dispatch(nullptr, request_header_, request_)));
+  EXPECT_EQ(callback_with_header_calls, 1);
+}
+
+TEST_F(TestAnyServiceCallback, set_and_dispatch_defered) {
```

Comment:
It would be great if an example was added to ros/examples for this new feature

Reply:
For anyone coming back to this later, I threw together an example of async services here: https://github.com/tgroechel/lifecycle_prac/blob/main/src/async_srv.cpp#L70-L108

Haven't compiled it yet but should give the idea (I patched this together while doing deferrable lifecycle transitions)

Reply:
And here is an example (also not compiled) leveraging the `SharedFutureResponse` in order to call a service within a service: https://github.com/tgroechel/lifecycle_prac/blob/main/src/async_srv.cpp#L10-L69C1